aesara/tensor/slinalg.py

```diff
@@ -192,57 +188,47 @@ def infer_shape(self, fgraph, node, shapes):
 
 class CholeskySolve(Op):
 
-    __props__ = ("lower", "check_finite")
-
-    def __init__(
-        self,
-        lower=True,
-        check_finite=True,
-    ):
-        self.lower = lower
-        self.check_finite = check_finite
-
-    def __repr__(self):
-        return "CholeskySolve{%s}" % str(self._props())
-
-    def make_node(self, C, b):
-        C = as_tensor_variable(C)
+    def make_node(self, a, b):
+        a = as_tensor_variable(a)
         b = as_tensor_variable(b)
-        assert C.ndim == 2
-        assert b.ndim in (1, 2)
-
-        # infer dtype by solving the most simple
-        # case with (1, 1) matrices
-        o_dtype = scipy.linalg.solve(
-            np.eye(1).astype(C.dtype), np.eye(1).astype(b.dtype)
-        ).dtype
-        x = tensor(shape=b.broadcastable, dtype=o_dtype)
-        return Apply(self, [C, b], [x])
-
-    def perform(self, node, inputs, output_storage):
-        C, b = inputs
-        rval = scipy.linalg.cho_solve(
-            (C, self.lower),
-            b,
-            check_finite=self.check_finite,
-        )
+        out_dtype = aes.upcast(a.dtype, b.dtype)
+        broadcastable = [
```

Comment:
We can do better than broadcastable. Have a look at https://aesara.readthedocs.io/en/latest/extending/type.html